co-log-core/src/Colog/Core/Action.hs

```diff
@@ -619,6 +620,34 @@ multiplicate (LogAction l) = LogAction $ \msgs -> l (fold msgs)
 {-# SPECIALIZE multiplicate :: Monoid msg => LogAction m msg -> LogAction m [msg] #-}
 {-# SPECIALIZE multiplicate :: Monoid msg => LogAction m msg -> LogAction m (NonEmpty msg) #-}
 
+{- | Like 'multiplicate' but instead of logging a batch of messages it logs each
+of them separately.
+
+>>> :{
+let logger :: LogAction IO [Int]
+    logger = logPrint
+in separate logger <& replicate 5 [1..3]
```

Comment:
Let's have a bit different example for easier understanding. Let's have `logger :: LogAction IO Int` and `separate logger <& [1..5]`. I think when types are `Int` and `[Int]` vs. `[Int]` and `[[Int]]`, it's a bit easier to understand what's going on in the example.